### PR TITLE
Add downward API docs

### PR DESCRIPTION
**Pull request by pmorie**
@adellape @bfallonf PTAL

### CODE REVIEW

**Comment by adellape** [comment]
Related card: https://trello.com/c/OqAvXb7d

**Comment by adellape** [comment]
@pmorie Overall LGTM. It's currently not in the build_cfg, so it won't show up in the Dev Guide yet. I assume this would be appropriate to appear immediately in both the Origin and Enterprise (3.0 GA) docs?

I think it could use a bit of context in the Overview as to when/why a user would want to use this. We could probably also add a really quick example at the end showing how to verify it's working, similar to https://github.com/GoogleCloudPlatform/kubernetes/blob/master/examples/downward-api/README.md#step-one-create-the-pod.

**Comment by pmorie** [comment]
@adellape Yep, I forget build_cfg again.  I will update it and add some context.

**Comment by pmorie** [comment]
@adellape PTAL

File: dev_guide/downward_api.adoc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+<1>  The pod's name
+<2>  The pod's namespace
+
+As an example:
+
+. Create a `*_pod.json_*` file:
+
+[source,yaml]
+---
```
Comment (adellape) [inline dev_guide/downward_api.adoc]: These (and the closing ones) need to be four hyphens for AsciiDoc.

**Comment by adellape** [comment]
@pmorie Just the one comment, otherwise LGTM.

**Comment by pmorie** [comment]
@adellape PTAL

**Comment by adellape** [comment]
:+1: Thanks.

**Pull request merged**